azure-pipelines.yml

```diff
@@ -2,72 +2,101 @@ trigger:
-        npm run lint 
-    displayName: 'npm lint'
+    - task: Npm@1
+      displayName: "npm Packaging"
+      inputs:
+          command: 'custom'
+          customCommand: 'pack'
+
+    - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
+      displayName: 'SBOM Generation Task'
+      inputs:
+        BuildDropPath: '$(Build.ArtifactStagingDirectory)'
+
+    - task: CopyFiles@2
+      inputs:
+        SourceFolder: '$(Build.SourcesDirectory)'
+        Contents: '*.tgz'
+        TargetFolder: '$(Build.ArtifactStagingDirectory)/Package'
 
-  - task: Npm@1
-    displayName: "npm Packaging"
-    inputs:
+    - task: PublishBuildArtifacts@1
+      inputs:
+        PathtoPublish: '$(Build.ArtifactStagingDirectory)/Package'
+        ArtifactName: 'drop'
+        publishLocation: 'Container'
+
+    - task: Npm@1
+      displayName: "npm Test"
+      inputs:
         command: 'custom'
-        customCommand: 'pack'
+        customCommand: 'test'
 
-  - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
-    displayName: 'SBOM Generation Task'
-    inputs:
-      BuildDropPath: '$(Build.ArtifactStagingDirectory)'
+    - task: GitHubRelease@1
+      condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
+      inputs:
+        gitHubConnection: 'github.com_marianan'
+        repositoryName: '$(Build.Repository.Name)'
+        action: 'create'
+        target: '$(Build.SourceVersion)'
+        tagSource: 'userSpecifiedTag'
+        tag: '$(Build.SourceBranch)'
+        releaseNotesSource: 'inline'
+        assets: '$(Build.ArtifactStagingDirectory)/Package/*'
+        isPreRelease: true
+        addChangeLog: false
 
-  - task: CopyFiles@2
-    inputs:
-      SourceFolder: '$(Build.SourcesDirectory)'
-      Contents: '*.tgz'
-      TargetFolder: '$(Build.ArtifactStagingDirectory)/Package'
-  - task: PublishBuildArtifacts@1
-    inputs:
-      PathtoPublish: '$(Build.ArtifactStagingDirectory)/Package'
-      ArtifactName: 'drop'
-      publishLocation: 'Container'
+    - task: Bash@3
+      inputs:
+        targetType: 'inline'
+        script: |
+          if [[ $(Build.SourceBranch) =~ ^refs/tags/v?[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+          echo "##vso[task.setvariable variable=NPM_PUBLISH_FLAG;]true"
+          fi
 
-  - task: Npm@1
-    displayName: "npm Test"
-    inputs:
-      command: 'custom'
-      customCommand: 'test'
-    
-  - task: GitHubRelease@1
-    condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
-    inputs:
-      gitHubConnection: 'github.com_marianan'
-      repositoryName: '$(Build.Repository.Name)'
-      action: 'create'
-      target: '$(Build.SourceVersion)'
-      tagSource: 'userSpecifiedTag'
-      tag: '$(Build.SourceBranch)'
-      releaseNotesSource: 'inline'
-      assets: '$(Build.ArtifactStagingDirectory)/Package/*'
-      addChangeLog: false
+    - task: Bash@3
```

Comment:
>task: Bash@3 [](http://example.com/codeflow?start=6&length=12)

Could we put this task behind NPM_PUBLISH_FLAG==true as we won't need it otherwise?

Reply:
yes